plugin.py

```diff
@@ -91,25 +91,30 @@ def weather(self, irc: callbacks.NestedCommandsIrcProxy, msg: ircmsgs.IrcMsg, ar
             else:
                 irc.reply("There is an error. Contact admin.", prefixNick=False)
 
-    @wrap([optional("int"), "text"])
+    @wrap([getopts({"metric": ""}), "text"])
     def setweather(
-        self, irc: callbacks.NestedCommandsIrcProxy, msg: ircmsgs.IrcMsg, args: List[str], num: int, text: str
+        self,
+        irc: callbacks.NestedCommandsIrcProxy,
+        msg: ircmsgs.IrcMsg,
+        args: List[str],
+        optlist: List[Tuple[str, bool]],
+        text: str,
     ) -> None:
-        """<display_format> <location>
-        Sets the weather location for a user.
-        Format number is set to 1 for imperial, and 2 for metric, for which
-        units to display first. e.g. setweather 2 70118
+        """[--metric] <location>
+        Sets the weather location for a user. Format is set to show imperial first by default.
+        To show metric first, use the --metric option. e.g. setweather --metric 70118
```

Comment:
Consider your users here.  You're changing the public interface without much warning.  You can either accept that and deal with the users directly when they screw up or you can add more code to inform them.  Both are completely reasonable paths forward, but you want to think ahead and at least have a mental plan.